internal/crypto/certgen.go

```diff
@@ -153,6 +154,11 @@ func GenerateCerts(cfg *config.Server) (*Certificates, error) {
 			return nil, err
 		}
 
+		oidcHMACSecret, err := generateHMACSecret()
```

Comment:
what happens during upgrade if this secret already exists ? do we skip or overwrite ?

Reply:
relates to https://github.com/envoyproxy/gateway/blob/ce1eb544d5e1b473ccd2c852be595d80fcf75ba8/internal/provider/kubernetes/secrets.go#L84

Reply:
The creation of HMAC secret will be skipped in `CreateOrUpdateSecrets` if it's already there, like the other certs. No changes are needed.